### PR TITLE
Log failed requests only

### PR DESCRIPTION
**Pull request by omerfirmak**

### CODE REVIEW

File: jsonrpc/server.go [hunk 1]
```diff
@@ -438,7 +431,10 @@ func (s *Server) handleRequest(ctx context.Context, req *Request) (*response, er
 	if errAny := tuple[1].Interface(); !isNil(errAny) {
 		res.Error = errAny.(*Error)
 		if res.Error.Code == InternalError {
-			s.listener.OnRequestFailed(req.Method, err)
+			s.listener.OnRequestFailed(req.Method, res.Error)
+			reqJSON, _ := json.Marshal(req)
+			errJSON, _ := json.Marshal(res.Error)
+			s.log.Debugw("failed handing RPC request", "req", string(reqJSON), "res", string(errJSON))
```
Comment (joshklop) [inline jsonrpc/server.go:437]:
```suggestion
			s.log.Debugw("Failed handing RPC request", "req", string(reqJSON), "res", string(errJSON))
```

Comment (joshklop) [inline jsonrpc/server.go:437]: If `reqJSON` or `errJSON` are `nil`, we can't convert them using `string(nil)`. I think we should set them to `[]byte{}` if `err != nil`.

Comment (omerfirmak) [reply jsonrpc/server.go:437]: Actually, `string([]byte(nil))` just returns an empty string. So we should be good.

File: jsonrpc/server.go [hunk 2]
```diff
@@ -438,7 +431,10 @@ func (s *Server) handleRequest(ctx context.Context, req *Request) (*response, er
 	if errAny := tuple[1].Interface(); !isNil(errAny) {
 		res.Error = errAny.(*Error)
 		if res.Error.Code == InternalError {
-			s.listener.OnRequestFailed(req.Method, err)
+			s.listener.OnRequestFailed(req.Method, res.Error)
+			reqJSON, _ := json.Marshal(req)
+			errJSON, _ := json.Marshal(res.Error)
+			s.log.Debugw("Failed handing RPC request", "req", string(reqJSON), "res", string(errJSON))
```
Comment (joshklop) [inline jsonrpc/server.go:437]: Might be good to promote this to a warning.

**Review by joshklop — APPROVED** [review]

File: jsonrpc/server.go (hunk 2 above)
Comment (wojciechos) [reply jsonrpc/server.go:437]: I would rather stay with debug. If we label it as a warning, we might start overlooking real warnings, especially if these pop up a lot.

**Pull request merged**